src/subsplit_dag.cpp

```diff
@@ -1082,43 +1112,36 @@ bool SubsplitDAG::IsConsistent() const {
 }
 
 bool SubsplitDAG::IsValid() const {
-  bool has_invalid_node = false;
-  IterateOverRealNodes([this, &has_invalid_node](SubsplitDAGNode node) {
-    if (has_invalid_node == false) {
-      if (node.IsValid() == false) {
-        has_invalid_node = true;
-      }
-    }
-  });
-  return has_invalid_node;
+  size_t correct_id = 0;
+  for (auto node : storage_.GetVertices()) {
+    if (correct_id++ != node.Id()) return false;
```

Comment:
current style guideline says the body of any conditional gets wrapped in braces. We can change that if  you feel strongly!

Reply:
This was from some rebasing by Ognian.  I am happy with or without braces.  We could ask Ognian if he has an opinion tomorrow.